components/usage/pkg/db/cost_center.go

```diff
@@ -23,3 +25,18 @@ type CostCenter struct {
 func (d *CostCenter) TableName() string {
 	return "d_b_cost_center"
 }
+
+func GetCostCenter(ctx context.Context, conn *gorm.DB, attributionId AttributionID) (*CostCenter, error) {
+	db := conn.WithContext(ctx)
+	var costCenter CostCenter
+
+	result := db.
+		Where("ID = ?", attributionId).
```

Comment:
Is the ID really capitalized in the DB?

You may be better of using the `db.First(&costCenter, attributionId)` function here

Reply:
> Is the ID really capitalized in the DB?

It's not. I was following the convention I saw in the other models in the package.

Reply:
Do you have an example? I have feeling it wouldn't actually work in this case as `ID` would need to be the exact name of the column (unless it does some case ignore)

Reply:
`d_b_project` is `id` lowercase in the DB but `ID` in the model. `d_b_billed_session` is `instanceId` but `InstanceID` in the model. I saw these and assumed it ignores cases.

Reply:
What matters for these are the annotations - the stuff in the backticks, look for something like 
```golang
`gorm:"column:usageAttributionId`
```

Reply:
The field names (as in regular go Fields) can be renamed however you like, but the column names can't (or would require a migration)

Reply:
> d_b_project is id lowercase in the DB but ID in the model. d_b_billed_session is instanceId but InstanceID in the model. I saw these and assumed it ignores cases.

In this case, it's `ID` as a field because the Go convention is to capitalize shorthands like id. But it still has lowercase `id` in the annotation, like `gorm:"primary_key;column:id;type:char;size:36;" json:"id"`

Reply:
Yess, the column annotations correspond to the columns

Reply:
Thanks good catch, the where clause was incorrect 🙌🏽 Have replaced it with the `db.First` syntax too.